### DISCUSSION TITLE
POTD, merge without extra space.

### DISCUSSION THREAD

**Discussion by VipinDhakad** [getlost01/gfg-potd#3] [Solution Ideas] [answered]
void swap(long long arr1[], long long arr2[], int j, int k){
            long long temp = arr2[j];
            arr2[j] = arr1[k];
            arr1[k] = temp;
            return;
        }
        void merge(long long arr1[], long long arr2[], int n, int m) 
        { 
            int i=0, j=0, k=n-1;
            
            while(i<k && j<m){
                if(arr1[i]>arr2[j]){
                    swap(arr1, arr2, j, k);
                    k--;
                    j++;
                }
                else i++;
            }
            sort(arr1, arr1+n);
            sort(arr2, arr2+m);
            return;
        }

**Comment by getlost01** [comment]
Thanks for your solution idea.
I have tried to run your code but it is giving a **WA**
<img width="800" alt="Screenshot 2023-07-07 at 2 51 45 PM" src="https://github.com/getlost01/gfg-potd/assets/79409258/8e058905-40dc-40bb-a6de-1cbe0fb2f724">

**Reply by VipinDhakad** [reply]
Thanks for reviewing, it's actually this condition -> (i<=k && j<m) in the while loop. I forgot to add a = sign while posting the answer here.

**Reply by getlost01** [reply, marked as answer]
Although your logic is correct, the problem still exists. This is because the problem states that to not use additional space, but your swap function uses a temporary variable `temp` multiple times(`n` times) in a loop for swapping. 

By the way, this question is just complex because of these `no extra space` constraints else it is just a merge sort concept.  

Thank you for reaching out and sharing your solution for diversifying solution ideas.
